File: bot_simple.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const espacios = '-------------------------------------';

function saludo() {
  console.log('BIENVENIDO A BOT.PosiVibe');
  console.log(espacios);
  console.log('Este bot te ayudará a motivarte o tener disciplina \n Hola, ¿cómo te llamas?');

  return new Promise((resolve) => {
    rl.question('Ingrese su nombre: ', (nombre) => {
      console.log(`Hola ${nombre}`);
      resolve(nombre);
    });
  });
}

function estadoAnimo() {
  console.log(espacios);
  return new Promise((resolve) => {
    rl.question('¿Cómo estás hoy? (bien/mal): ', (estadoDeAnimo) => {
      if (estadoDeAnimo.toLowerCase() === 'bien') {
        console.log('¡Qué bueno que estés bien :)');
      } else if (estadoDeAnimo.toLowerCase() === 'mal') {
        console.log('Qué mal, si quieres te recomiendo una página de psicología');
      } else {
        console.log('Gracias por compartir tu estado de ánimo :)');
      }
      resolve();
    });
  });
}

function recomendacionPagina() {
  console.log(espacios);
  return new Promise((resolve) => {
    rl.question('¿Quieres una página que te puede motivar y mejorar? (si/no): ', (paginaRecomendada) => {
      if (paginaRecomendada.toLowerCase() === 'si') {
        console.log('Aquí está el Link: https://www.uv.mx/veracruz/cess/vinculacion-y-extension/psicologia/');
      } else if (paginaRecomendada.toLowerCase() === 'no') {
        console.log('Ok, que tengas bonito día :)');
      } else {
        console.log("Por favor responda con 'si' o 'no'.");
      }
      resolve();
    });
  });
}

function fraseMotivacional() {
  console.log(espacios);
  return new Promise((resolve) => {
    rl.question('¿Crees que necesitas una frase motivacional? (si/no): ', (fraseMotivacional) => {
      if (fraseMotivacional.toLowerCase() === 'si') {
        console.log('El éxito es la suma de pequeños esfuerzos repetidos día tras día.');
      } else if (fraseMotivacional.toLowerCase() === 'no') {
        console.log('Entiendo, estoy aquí si quieres hablar de otra cosa.');
      } else {
        console.log("Por favor responda con 'si' o 'no'.");
      }
      resolve();
    });
  });
}

function recomendacionLibros() {
  console.log(espacios);
  const libros = [
    '1. Si lo crees, lo creas',
    '2. Hábitos atómicos',
    '3. Maestría'
  ];

  function mostrarLibros() {
    console.log(espacios);
    console.log('Aquí tienes algunos libros que podrías leer y mejorar:');
    libros.forEach((libro) => console.log(libro));
  }

  mostrarLibros();

  return new Promise((resolve) => {
    rl.question('¿Te gustaría recibir una recomendación de un libro? (si/no): ', (respuesta) => {
      if (respuesta.toLowerCase() === 'si') {
        const libroRecomendado = libros[Math.floor(Math.random() * libros.length)];
        console.log(`Aquí tienes una recomendación: ${libroRecomendado}`);
      } else if (respuesta.toLowerCase() === 'no') {
        console.log('Ok, está bien. Si cambias de opinión, aquí estaré.');
      } else {
        console.log("Por favor, responde con 'si' o 'no'.");
      }
      resolve();
    });
  });
}

function estadoAnimo2() {
  console.log(espacios);
  return new Promise((resolve) => {
    rl.question('¿Crees que ha mejorado tu estado de ánimo? (si/no): ', (preguntaEstadoAnimo) => {
      if (preguntaEstadoAnimo.toLowerCase() === 'si') {
        console.log('Excelente, sigue mejorando :)');
      } else if (preguntaEstadoAnimo.toLowerCase() === 'no') {
        console.log('Creo que necesitas ver una película o escuchar música');
      } else {
        console.log("Por favor responda con 'si' o 'no'.");
      }
      resolve();
    });
  });
}

function recomendacionPeliculas() {
  console.log(espacios);
  const peliculas = [
    '1. Una mente maravillosa',
    '2. Un sueño posible',
    '3. En busca de la felicidad'
  ];

  function mostrarPelicula() {
    console.log(espacios);
    console.log('Aquí tienes algunas películas interesantes:');
    peliculas.forEach((pelicula) => console.log(pelicula));
  }

  mostrarPelicula();

  return new Promise((resolve) => {
    rl.question('¿Quieres una recomendación de alguna película? (si/no): ', (recomendacionPeliculas) => {
      if (recomendacionPeliculas.toLowerCase() === 'si') {
        const peliculaRecomendada = peliculas[Math.floor(Math.random() * peliculas.length)];
        console.log(`Aquí tienes una recomendación: ${peliculaRecomendada}`);
      } else if (recomendacionPeliculas.toLowerCase() === 'no') {
        console.log('Ok, si cambias de opinión aquí estaré.');
      } else {
        console.log("Por favor responda con 'si' o 'no'.");
      }
      resolve();
    });
  });
}

function recomendacionRutina() {
  console.log(espacios);
  const rutinas = [
    '1. Dormir temprano',
    '2. Levantarse temprano',
    '3. Mantener una dieta saludable',
    '4. Haz una lista de tus objetivos y metas',
    '5. Haz algún tipo de entrenamiento físico',
    '6. Lee un libro de crecimiento personal',
    '7. Mantente alejado de algún vicio',
    '8. Trata de aprender algo nuevo cada día',
    '9. Crea un grupo de amigos que te ayuden a ser mejor persona',
    '10. Practica la meditación dos veces al día'
  ];

  function mostrarRutina() {
    console.log(espacios);
    console.log('Aquí tienes una rutina efectiva:');
    rutinas.forEach((rutina) => console.log(rutina));
  }

  mostrarRutina();

  return new Promise((resolve) => {
    rl.question('¿Quieres una rutina efectiva? (si/no): ', (recomendacionRutina) => {
      if (recomendacionRutina.toLowerCase() === 'si') {
        const rutinaRecomendada = rutinas[Math.floor(Math.random() * rutinas.length)];
        console.log('La rutina del éxito');
      } else if (recomendacionRutina.toLowerCase() === 'no') {
        console.log('Ok, entiendo. Si cambias de opinión, aquí estaré.');
      } else {
        console.log("Por favor responde con 'si' o 'no'.");
      }
      resolve();
    });
  });
}

async function main() {
  await saludo();
  await estadoAnimo();
  await recomendacionPagina();
  await fraseMotivacional();
  await recomendacionLibros();
  await estadoAnimo2();
  await recomendacionPeliculas();
  await recomendacionRutina();
  console.log(espacios);
  console.log('Aquí te podemos ayudar a mejorar :)');
  console.log(espacios);
  rl.close();
}

main();

File: bot_simple.py
import random

espacios = "-------------------------------------"
def saludo():
 print("BIENVENIDO A BOT.PosiVibe") 
 print(espacios)
 print("este bot te ayudara a motivarte o tener disciplina \n hola, como te llamas?")
 nombre = input("Ingrese su nombre: ")
 print(f"Hola {nombre} ")
 return nombre
print(espacios)

def estado_animo():
 print(espacios)
 estado_de_animo = input("Como estas hoy? (bien/mal): ")
 animo_1 = "bien"
 animo_2 = "mal"
 if estado_de_animo.lower() == animo_1:
    print("Que bueno que estes bien :)")
 elif estado_de_animo.lower() == animo_2.lower():
    print("Que mal, si quieres te recomiendo una pagina de psicología")
 else:
    print("Gracias por compartir tu estado de animo :)")

def recomendacion_pagina():
 print(espacios)
 pagina_recomendada = input("Quieres una pagina que te puede motivar y mejorar? (si/no): ")
 respuesta_1 = "si"
 respuesta_2 = "no"

 if pagina_recomendada.lower() == respuesta_1:
    print("Aqui esta el Link: https://www.uv.mx/veracruz/cess/vinculacion-y-extension/psicologia/")
 elif pagina_recomendada.lower() == respuesta_2:
    print("ok que tengas bonito dia :)")
 else:
    print("Por favor responda con 'si' o 'no'. " )

def frase_motivacional():
 print(espacios)
 frase_motivacional = input("Crees que necesistes una frase motivacional? (si/no): ")
 respuesta_3 = "si"
 respuesta_4 = "no"
 if frase_motivacional.lower() == respuesta_3:
    print("El éxito es la suma de pequeños esfuerzos repetidos día tras día.")
 elif frase_motivacional.lower() == respuesta_4:
    print("Entiendo, estoy aquí si quieres hablar de otra cosa")
 else:
    print("Por favor responda con 'si' o 'no'. ")

def recomendacion_libros():
 print(espacios)
 respuesta = input("¿Te gustaría recibir una recomendación de un libro? (si/no): ")
 respuesta_5 = "si"
 respuesta_6 = "no"
 libros = [
    "1. Si lo crees, lo creas",
    "2. Hábitos atómicos",
    "3. Maestría"
]

 def mostrar_libros():
    print(espacios)
    print("Aquí tienes algunos libros que podrías leer y mejorar:")
    for libro in libros:
        print(libro)

 mostrar_libros()

 if respuesta.lower() == respuesta_5:
  libro_recomendado = random.choice(libros)
  print(f"Aquí tienes una recomendación: {libro_recomendado}")
 elif respuesta.lower() == respuesta_6:
    print("Ok, está bien. Si cambias de opinión, aquí estaré.")
 else:
    print("Por favor, responde con 'si' o 'no'.")

def estado_animo_2():
 print(espacios)
 pregunta_estado_animo = input("Crees que ha mejorado tu estado de animo? (si/no):")  
 
 respuesta_7 = "si"
 respuesta_8 = "no"

 if pregunta_estado_animo.lower() == respuesta_7:
   print("Exelente sigue mejorando :)")
 elif pregunta_estado_animo.lower() == respuesta_8:
   print("Creo que necesitas ver una pelicula o escuchar musica")
 else:
   print("Por favor responda con ('si' o 'no'): ")

def recomendacion_peliculas():
 print(espacios)
 recomendacion_peliculas = input("quieres una recomendacion de alguna pelicula? (si/no):")
 
 respuesta_9 = "si"
 respuesta_10 = "mo"

 peliculas = [
   "1. Una mente maravillosa",
   "2. Un sueño posible ",
   "3. En busca de la felicidad"
 ]
 def mostrar_pelicula():
   print(espacios)
   print("Aqui tienes algunas peliculas interesantes:")
   for pelicula in peliculas:
     print(pelicula)
     
 mostrar_pelicula()
 if recomendacion_peliculas.lower() == respuesta_9:
    pelicula_recomendada = random.choice(peliculas)
    print(f"Aqui tienes una recomendacion: {pelicula_recomendada}")
 elif recomendacion_peliculas.lower() == respuesta_10:
   print("OK si cambias de opinion aqui estare.")
 else:
   print("Por favor responda con 'si' o 'no'.")

def recomendacion_rutina():
 print(espacios)
 recomendacion_rutina = input("Quieres una rutina efectiva? (si/no): ")

 respuesta_11 = "si"
 respuesta_12 = "no"

 rutinas = [
    "1. dormir temprano ",
    "2. levantarse temprano",
    "3. Mantener una dieta saludable ",
    "4. haz una lista de tus objetivos y metas",
    "5. haz algun tipo de entrenamiento fisico",
    "6. lee un libro de crecimiento personal"
    "7. mantente alejado de algun vicio ",
    "8. trata de aprender algo nuevo cada dia",
    "9. crea un grupo de amigos que te ayuden a ser mejor persona",
    "10. practica la meditacion dos veces al dia"
    
  ]
 def mostrar_rutina():
    print(espacios)
    print("Aqui tienes una rutina efectiva: ")
    for rutina in rutinas:
      print(rutina)

 mostrar_rutina()
 if recomendacion_rutina.lower() == respuesta_11:
     rutina_recomendada = random.choice(rutinas)
     print("La rutina del exito")
 elif recomendacion_rutina.lower() == respuesta_12:
    print("OK entiendo si cambias de opinion aqui estare.")
 else:
    print("Por favor responde con 'si' o 'no'.")




def main():
   saludo()
   estado_animo()
   recomendacion_pagina()
   frase_motivacional()
   recomendacion_libros()
   estado_animo_2()
   recomendacion_peliculas()
   recomendacion_rutina()
   print(espacios)
   print("Aquí te podemos ayudar a mejorar :) .")
   print(espacios)

if __name__ == "__main__":
    main()







